### ISSUE TITLE
google_checks SuppressionXpathSingleFilter for 'MethodName' check is too lenient in several ways

### ISSUE THREAD

**Issue by mches** [checkstyle/checkstyle#15609]
I have read check documentation: https://checkstyle.org/filters/suppressionxpathsinglefilter.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac PrintStreamTest.java

/var/tmp $ cat /google_checks.xml
cat: /google_checks.xml: No such file or directory

/var/tmp $ cat PrintStreamTest.java
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

class PrintStreamTest {
  void println__String_(@Test String x) {
    class Local {
      void println__String_(String x) {
        System.out.println(x);
      }
    }

    new Local().println__String_(x);
  }

  @Target(ElementType.PARAMETER)
  @Retention(RetentionPolicy.RUNTIME)
  @Documented
  @interface Test {}
}

/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c /google_checks.xml PrintStreamTest.java
Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**
The two illegally-named non-test methods should be reported e.g.

```
Starting audit...
[WARN] /var/tmp/PrintStreamTest.java:8:8: Method name 'println__String_' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
[WARN] /var/tmp/PrintStreamTest.java:10:12: Method name 'println__String_' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
Audit done.
```

1. Multiple consecutive underscores are not allowed for test methods.
2. Trailing underscores are not allowed for test methods.
3. Each underscore-separated component should start with a lowercase letter.
4. Without a method annotation, the method is not a test method. 
5. A method on a local or anonymous class declared in a test method is not necessarily a test method.

If approved, [here](https://github.com/markitect-dev/checkstyle/commit/641ed74715809f7e6af20f2eefdf027e3d3e6a29?diff=unified&w=0) is a proposal I would like to submit as a PR.

Before:
```xml
    <module name="SuppressionXpathSingleFilter">
      <property name="checks" value="MethodName"/>
      <property name="query" value="//METHOD_DEF[
                                     .//ANNOTATION/IDENT[contains(@text, 'Test')]]//*"/>
      <property name="message" value="'[a-z][a-z0-9][_a-zA-Z0-9]*'.*"/>
    </module>
```
↓↓↓

After:
```xml
    <module name="SuppressionXpathSingleFilter">
      <property name="checks" value="MethodName"/>
      <property name="query" value="//METHOD_DEF[
                                     ./MODIFIERS/ANNOTATION//IDENT[contains(@text, 'Test')]]/IDENT"/>
      <property name="message" value="'[a-z][a-z0-9][a-zA-Z0-9]*(?:_[a-z][a-z0-9][a-zA-Z0-9]*)*'"/>
    </module>
```

Note this proposal also accommodates fully-qualified test annotations. i.e.

```bash
/var/tmp $ javac -cp apiguardian-api-1.1.2.jar:junit-jupiter-api-5.11.0.jar Test.java

/var/tmp $ cat /google_checks.xml
cat: /google_checks.xml: No such file or directory

/var/tmp $ cat Test.java
class Test {
  @org.junit.jupiter.api.Test
  void underscoreSeparated_lowerCamelCase_components() {}
}

/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.18.1-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:3:8: Method name 'underscoreSeparated_lowerCamelCase_components' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'. [MethodName]
Audit done.
```

**Comment by romani** [comment]
>Still not clear ...

Should be removed from issue description

**Comment by Zopsss** [comment]
> 3. Each underscore-separated component should start with a lowercase letter.

We should allow digits immediately after underscore, reason explained at: https://github.com/checkstyle/checkstyle/pull/15617#discussion_r1744093928

**Comment by mches** [comment]
> > 3. Each underscore-separated component should start with a lowercase letter.
> 
> 
> 
> We should allow digits immediately after underscore, reason explained at: https://github.com/checkstyle/checkstyle/pull/15617#discussion_r1744093928

I'll have a look. Thanks for the thorough review

**Comment by romani** [comment]
Fix is merged
